tests/functional/watcher_controller_test.go

```diff
@@ -371,4 +387,32 @@ var _ = Describe("Watcher controller", func() {
 		})
 	})
 
+	When("Watcher is created  with no container images defined and env variables contains fake value", func() {
+		BeforeEach(func() {
+			DeferCleanup(th.DeleteInstance, CreateWatcher(watcherTest.Instance, MinimalWatcherContainerSpec))
+			// Set environment variables
+			os.Setenv("RELATED_IMAGE_WATCHER_API_IMAGE_URL_DEFAULT", "watcher-api-custom-image")
```

Comment:
You are setting the same values in env variables that the ones on MinimalWatcherContainerSpec so in this case to guess if the real parameter is coming from env variable or from the spec. I'd set different values for env (mabye adding -env or something.)